### ISSUE TITLE
struct return - cannot find deal with ptr that isnt arg

### ISSUE THREAD

**Issue by ZuseZ4** [EnzymeAD/Enzyme#837]
cpp Source: openLB dec7cc8f9
related?:  https://github.com/EnzymeAD/Enzyme/pull/815/files

Will try to create some mre based on `_ZN3olb7forcing3GuoINS_7momenta6ForcedEE18combined_collisionINS_11descriptors5D3Q19IJNS6_5FORCEEEEENS2_5TupleINS2_11BulkDensityENS2_12BulkMomentumENS2_10BulkStressENS2_11DefineToNEqEEENS_10equilibria11SecondOrderENS_9collision3BGKEE5applyINS_4CellIdS9_EENS_11ParametersDIdS9_JNS6_5OMEGAEEEEdEENS_13CellStatisticIT1_EERT_RT0_` ?

```.cpp
oval=  %17 = call { double, double } @_ZN3olb7forcing3GuoINS_7momenta6ForcedEE18combined_collisionINS_11descriptors5D3Q19IJNS6_5FORCEEEEENS2_5TupleINS2_11BulkDensityENS2_12BulkMomentumENS2_10BulkStressENS2_11DefineToNEqEEENS_10equilibria11SecondOrderENS_9collision3BGKEE5applyINS_4CellIdS9_EENS_11ParametersDIdS9_JNS6_5OMEGAEEEEdEENS_13CellStatisticIT1_EERT_RT0_(%"class.std::ios_base::Init"* noundef nonnull align 1 dereferenceable(1) %6, %"class.olb::Cell"* noundef nonnull align 8 dereferenceable(168) %1, %"struct.olb::ParametersD.694"* noundef nonnull align 8 dereferenceable(16) %4) #43 icv=0
available inversion for   %9 = load %"struct.olb::ParametersD.694"*, %"struct.olb::ParametersD.694"** %8, align 8, !tbaa !2799 of   %16 = load %"struct.olb::ParametersD.694"*, %"struct.olb::ParametersD.694"** %14, align 8, !tbaa !2799
available inversion for   %4 = bitcast i8* %3 to %"struct.olb::ParametersD.694"* of   %7 = bitcast i8* %6 to %"struct.olb::ParametersD.694"*
available inversion for @_ZTVN3olb11ParametersDIdNS_11descriptors5D3Q19IJNS1_5FORCEEEEEJNS1_5OMEGAEEEE = internal unnamed_addr constant { [4 x i8*] } { [4 x i8*] [i8* null, i8* bitcast ({ i8*, i8*, i32, i32, i8*, i64, i8*, i64 }* @_ZTIN3olb11ParametersDIdNS_11descriptors5D3Q19IJNS1_5FORCEEEEEJNS1_5OMEGAEEEE to i8*), i8* bitcast (void (%"class.std::basic_ostream.base"*)* @_ZN3olb18AbstractParametersIdNS_11descriptors5D3Q19IJNS1_5FORCEEEEEED2Ev to i8*), i8* bitcast (void (%"struct.olb::ParametersD.694"*)* @_ZN3olb11ParametersDIdNS_11descriptors5D3Q19IJNS1_5FORCEEEEEJNS1_5OMEGAEEED0Ev to i8*)] }, comdat, align 8, !type !268, !type !239, !enzyme_shadow !269 of @_ZTVN3olb11ParametersDIdNS_11descriptors5D3Q19IJNS1_5FORCEEEEEJNS1_5OMEGAEEEE_shadow = internal unnamed_addr constant { [4 x i8*] } { [4 x i8*] [i8* null, i8* bitcast ({ i8*, i8*, i32, i32, i8*, i64, i8*, i64 }* @_ZTIN3olb11ParametersDIdNS_11descriptors5D3Q19IJNS1_5FORCEEEEEJNS1_5OMEGAEEEE to i8*), i8* bitcast ({ i8* (%"class.std::basic_ostream.base"*, %"class.std::basic_ostream.base"*)*, void (%"class.std::basic_ostream.base"*, %"class.std::basic_ostream.base"*, i8*)* }* @"_enzyme_reverse__ZN3olb18AbstractParametersIdNS_11descriptors5D3Q19IJNS1_5FORCEEEEEED2Ev'" to i8*), i8* bitcast ({ i8* (%"struct.olb::ParametersD.694"*, %"struct.olb::ParametersD.694"*)*, void (%"struct.olb::ParametersD.694"*, %"struct.olb::ParametersD.694"*, i8*)* }* @"_enzyme_reverse__ZN3olb11ParametersDIdNS_11descriptors5D3Q19IJNS1_5FORCEEEEEJNS1_5OMEGAEEED0Ev'" to i8*)] }, align 8
available inversion for   %6 = bitcast i8* %5 to %"class.std::ios_base::Init"* of   %11 = bitcast i8* %10 to %"class.std::ios_base::Init"*
available inversion for   %5 = tail call noalias nonnull dereferenceable(1) dereferenceable_or_null(1) i8* @malloc(i64 1), !enzyme_fromstack !2798 of   %10 = tail call noalias nonnull dereferenceable(1) dereferenceable_or_null(1) i8* @malloc(i64 1)
available inversion for   %10 = getelementptr inbounds %"struct.olb::ParametersD.694", %"struct.olb::ParametersD.694"* %4, i64 0, i32 0, i32 0 of   %18 = getelementptr inbounds %"struct.olb::ParametersD.694", %"struct.olb::ParametersD.694"* %7, i64 0, i32 0, i32 0
available inversion for   %8 = getelementptr inbounds %"class.olb::dynamics::Tuple.1227", %"class.olb::dynamics::Tuple.1227"* %0, i64 0, i32 1 of   %14 = getelementptr inbounds %"class.olb::dynamics::Tuple.1227", %"class.olb::dynamics::Tuple.1227"* %1, i64 0, i32 1
available inversion for %"class.olb::Cell"* %1 of %"class.olb::Cell"* %3
available inversion for %"class.olb::dynamics::Tuple.1227"* %0 of %"class.olb::dynamics::Tuple.1227"* %1
available inversion for   %3 = tail call noalias nonnull dereferenceable(16) dereferenceable_or_null(16) i8* @malloc(i64 16), !enzyme_fromstack !2797 of   %6 = tail call noalias nonnull dereferenceable(16) dereferenceable_or_null(16) i8* @malloc(i64 16)
ld.lld: /home/drehwald/prog/Enzyme/enzyme/Enzyme/GradientUtils.cpp:4871: llvm::Value *GradientUtils::invertPointerM(llvm::Value *const, IRBuilder<> &, bool): Assertion `0 && "cannot find deal with ptr that isnt arg"' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /home/drehwald/prog/llvm-14/build/bin/ld.lld -z relro --hash-style=gnu --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o tFS_t /lib/x86_64-linux-gnu/crt1.o /lib/x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/11/crtbegin.o -L../../../external/lib -L/usr/lib/gcc/x86_64-linux-gnu/11 -L/usr/lib/gcc/x86_64-linux-gnu/11/../../../../lib64 -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib64 -L/lib -L/usr/lib -plugin-opt=mcpu=cascadelake -plugin-opt=O3 testFlowSolver.o -lz -ltinyxml -lz -ltinyxml --lto-legacy-pass-manager -mllvm=-load=/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so -mllvm=-enzyme-strict-aliasing=0 -mllvm=-enzyme-loose-types -mllvm=-enzyme-type-warning=0 -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/11/crtend.o /lib/x86_64-linux-gnu/crtn.o
1.	Running pass 'Enzyme Pass' on module 'ld-temp.o'.
 #0 0x00000000015c4fe3 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x15c4fe3)
 #1 0x00000000015c2dbe llvm::sys::RunSignalHandlers() (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x15c2dbe)
 #2 0x00000000015c55fa SignalHandler(int) Signals.cpp:0:0
 #3 0x00007fdf21c8d520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #4 0x00007fdf21ce1a7c __pthread_kill_implementation ./nptl/./nptl/pthread_kill.c:44:76
 #5 0x00007fdf21ce1a7c __pthread_kill_internal ./nptl/./nptl/pthread_kill.c:78:10
 #6 0x00007fdf21ce1a7c pthread_kill ./nptl/./nptl/pthread_kill.c:89:10
 #7 0x00007fdf21c8d476 gsignal ./signal/../sysdeps/posix/raise.c:27:6
 #8 0x00007fdf21c737f3 abort ./stdlib/./stdlib/abort.c:81:7
 #9 0x00007fdf21c7371b _nl_load_domain ./intl/./intl/loadmsgcat.c:1177:9
#10 0x00007fdf21c84e96 (/lib/x86_64-linux-gnu/libc.so.6+0x39e96)
#11 0x00007fdf1fd0f41d GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x30941d)
#12 0x00007fdf1fbb7e00 EnzymeLogic::CreateAugmentedPrimal(llvm::Function*, DIFFE_TYPE, llvm::ArrayRef<DIFFE_TYPE>, TypeAnalysis&, bool, bool, FnTypeInfo const&, std::map<llvm::Argument*, bool, std::less<llvm::Argument*>, std::allocator<std::pair<llvm::Argument* const, bool> > >, bool, unsigned int, bool, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x1b1e00)
#13 0x00007fdf1fd23ce4 GradientUtils::GetOrCreateShadowFunction(EnzymeLogic&, llvm::TargetLibraryInfo&, TypeAnalysis&, llvm::Function*, DerivativeMode, unsigned int, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x31dce4)
#14 0x00007fdf1fd0c949 GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306949)
#15 0x00007fdf1fd0cc7b GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306c7b)
#16 0x00007fdf1fd0b6d2 GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x3056d2)
#17 0x00007fdf1fd0bade GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x305ade)
#18 0x00007fdf1fd0ce37 GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306e37)
#19 0x00007fdf1fd0cc7b GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306c7b)
#20 0x00007fdf1fd0cc7b GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306c7b)
#21 0x00007fdf1fc144f0 AdjointGenerator<AugmentedReturn*>::visitCommonStore(llvm::Instruction&, llvm::Value*, llvm::Value*, llvm::MaybeAlign, bool, llvm::AtomicOrdering, unsigned char, llvm::Value*) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x20e4f0)
#22 0x00007fdf1fc1369a AdjointGenerator<AugmentedReturn*>::visitStoreInst(llvm::StoreInst&) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x20d69a)
#23 0x00007fdf1fbb7712 EnzymeLogic::CreateAugmentedPrimal(llvm::Function*, DIFFE_TYPE, llvm::ArrayRef<DIFFE_TYPE>, TypeAnalysis&, bool, bool, FnTypeInfo const&, std::map<llvm::Argument*, bool, std::less<llvm::Argument*>, std::allocator<std::pair<llvm::Argument* const, bool> > >, bool, unsigned int, bool, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x1b1712)
#24 0x00007fdf1fd23ce4 GradientUtils::GetOrCreateShadowFunction(EnzymeLogic&, llvm::TargetLibraryInfo&, TypeAnalysis&, llvm::Function*, DerivativeMode, unsigned int, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x31dce4)
#25 0x00007fdf1fd0c949 GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306949)
#26 0x00007fdf1fc144f0 AdjointGenerator<AugmentedReturn*>::visitCommonStore(llvm::Instruction&, llvm::Value*, llvm::Value*, llvm::MaybeAlign, bool, llvm::AtomicOrdering, unsigned char, llvm::Value*) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x20e4f0)
#27 0x00007fdf1fc1369a AdjointGenerator<AugmentedReturn*>::visitStoreInst(llvm::StoreInst&) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x20d69a)
#28 0x00007fdf1fbb7712 EnzymeLogic::CreateAugmentedPrimal(llvm::Function*, DIFFE_TYPE, llvm::ArrayRef<DIFFE_TYPE>, TypeAnalysis&, bool, bool, FnTypeInfo const&, std::map<llvm::Argument*, bool, std::less<llvm::Argument*>, std::allocator<std::pair<llvm::Argument* const, bool> > >, bool, unsigned int, bool, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x1b1712)
#29 0x00007fdf1fc2860f AdjointGenerator<AugmentedReturn*>::visitCallInst(llvm::CallInst&) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x22260f)
#30 0x00007fdf1fbb7712 EnzymeLogic::CreateAugmentedPrimal(llvm::Function*, DIFFE_TYPE, llvm::ArrayRef<DIFFE_TYPE>, TypeAnalysis&, bool, bool, FnTypeInfo const&, std::map<llvm::Argument*, bool, std::less<llvm::Argument*>, std::allocator<std::pair<llvm::Argument* const, bool> > >, bool, unsigned int, bool, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x1b1712)
#31 0x00007fdf1fd23ce4 GradientUtils::GetOrCreateShadowFunction(EnzymeLogic&, llvm::TargetLibraryInfo&, TypeAnalysis&, llvm::Function*, DerivativeMode, unsigned int, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x31dce4)
#32 0x00007fdf1fd0c949 GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306949)
#33 0x00007fdf1fd0cc7b GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306c7b)
#34 0x00007fdf1fd0b6d2 GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x3056d2)
#35 0x00007fdf1fd0bade GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x305ade)
#36 0x00007fdf1fd0ce37 GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306e37)
#37 0x00007fdf1fd0cc7b GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306c7b)
#38 0x00007fdf1fd0cc7b GradientUtils::invertPointerM(llvm::Value*, llvm::IRBuilder<llvm::ConstantFolder, llvm::IRBuilderDefaultInserter>&, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x306c7b)
#39 0x00007fdf1fc144f0 AdjointGenerator<AugmentedReturn*>::visitCommonStore(llvm::Instruction&, llvm::Value*, llvm::Value*, llvm::MaybeAlign, bool, llvm::AtomicOrdering, unsigned char, llvm::Value*) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x20e4f0)
#40 0x00007fdf1fc1369a AdjointGenerator<AugmentedReturn*>::visitStoreInst(llvm::StoreInst&) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x20d69a)
#41 0x00007fdf1fbb7712 EnzymeLogic::CreateAugmentedPrimal(llvm::Function*, DIFFE_TYPE, llvm::ArrayRef<DIFFE_TYPE>, TypeAnalysis&, bool, bool, FnTypeInfo const&, std::map<llvm::Argument*, bool, std::less<llvm::Argument*>, std::allocator<std::pair<llvm::Argument* const, bool> > >, bool, unsigned int, bool, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x1b1712)
#42 0x00007fdf1fc2860f AdjointGenerator<AugmentedReturn*>::visitCallInst(llvm::CallInst&) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x22260f)
#43 0x00007fdf1fbb7712 EnzymeLogic::CreateAugmentedPrimal(llvm::Function*, DIFFE_TYPE, llvm::ArrayRef<DIFFE_TYPE>, TypeAnalysis&, bool, bool, FnTypeInfo const&, std::map<llvm::Argument*, bool, std::less<llvm::Argument*>, std::allocator<std::pair<llvm::Argument* const, bool> > >, bool, unsigned int, bool, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x1b1712)
#44 0x00007fdf1fc2860f AdjointGenerator<AugmentedReturn*>::visitCallInst(llvm::CallInst&) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x22260f)
#45 0x00007fdf1fbb7712 EnzymeLogic::CreateAugmentedPrimal(llvm::Function*, DIFFE_TYPE, llvm::ArrayRef<DIFFE_TYPE>, TypeAnalysis&, bool, bool, FnTypeInfo const&, std::map<llvm::Argument*, bool, std::less<llvm::Argument*>, std::allocator<std::pair<llvm::Argument* const, bool> > >, bool, unsigned int, bool, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x1b1712)
#46 0x00007fdf1fc90dff AdjointGenerator<AugmentedReturn const*>::visitCallInst(llvm::CallInst&) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x28adff)
#47 0x00007fdf1fbc94c3 EnzymeLogic::CreatePrimalAndGradient(ReverseCacheKey const&&, TypeAnalysis&, AugmentedReturn const*, bool) (/home/drehwald/prog/Enzyme/enzyme/buildRel/Enzyme/LLDEnzyme-14.so+0x1c34c3)
#48 0x00007fdf1fb9eb59 (anonymous namespace)::Enzyme::HandleAutoDiff(llvm::CallInst*, llvm::TargetLibraryInfo&, DerivativeMode, bool) Enzyme.cpp:0:0
#49 0x00007fdf1fb99515 (anonymous namespace)::Enzyme::lowerEnzymeCalls(llvm::Function&, bool&, std::set<llvm::Function*, std::less<llvm::Function*>, std::allocator<llvm::Function*> >&) Enzyme.cpp:0:0
#50 0x00007fdf1fb95187 (anonymous namespace)::Enzyme::runOnModule(llvm::Module&) Enzyme.cpp:0:0
#51 0x00000000036aef6c llvm::legacy::PassManagerImpl::run(llvm::Module&) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x36aef6c)
#52 0x00000000020bfb0f llvm::lto::opt(llvm::lto::Config const&, llvm::TargetMachine*, unsigned int, llvm::Module&, bool, llvm::ModuleSummaryIndex*, llvm::ModuleSummaryIndex const*, std::vector<unsigned char, std::allocator<unsigned char> > const&) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x20bfb0f)
#53 0x00000000020c1ed1 llvm::lto::backend(llvm::lto::Config const&, std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream> > > (unsigned int)>, unsigned int, llvm::Module&, llvm::ModuleSummaryIndex&) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x20c1ed1)
#54 0x00000000020b5587 llvm::lto::LTO::runRegularLTO(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream> > > (unsigned int)>) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x20b5587)
#55 0x00000000020b4bec llvm::lto::LTO::run(std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream> > > (unsigned int)>, std::function<llvm::Expected<std::function<llvm::Expected<std::unique_ptr<llvm::CachedFileStream, std::default_delete<llvm::CachedFileStream> > > (unsigned int)> > (unsigned int, llvm::StringRef)>) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x20b4bec)
#56 0x000000000173e75c lld::elf::BitcodeCompiler::compile() (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x173e75c)
#57 0x000000000169a736 void lld::elf::LinkerDriver::compileBitcodeFiles<llvm::object::ELFType<(llvm::support::endianness)1, true> >(bool) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x169a736)
#58 0x000000000169737f lld::elf::LinkerDriver::link(llvm::opt::InputArgList&) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x169737f)
#59 0x000000000168bcdf lld::elf::LinkerDriver::linkerMain(llvm::ArrayRef<char const*>) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x168bcdf)
#60 0x000000000168a0bf lld::elf::link(llvm::ArrayRef<char const*>, llvm::raw_ostream&, llvm::raw_ostream&, bool, bool) (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x168a0bf)
#61 0x0000000001519a62 lldMain(int, char const**, llvm::raw_ostream&, llvm::raw_ostream&, bool) lld.cpp:0:0
#62 0x0000000001519257 main (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x1519257)
#63 0x00007fdf21c74d90 __libc_start_call_main ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
#64 0x00007fdf21c74e40 call_init ./csu/../csu/libc-start.c:128:20
#65 0x00007fdf21c74e40 __libc_start_main ./csu/../csu/libc-start.c:379:5
#66 0x0000000001518d95 _start (/home/drehwald/prog/llvm-14/build/bin/ld.lld+0x1518d95)
clang-14: error: unable to execute command: Aborted (core dumped)
clang-14: error: linker command failed due to signal (use -v to see invocation)
make: *** [Makefile:66: test] Error 254
```

edit: is this one another issue? There are a few of these.
`ld.lld: warning: <unknown>:0:0: could not statically determine size of realloc   %21 = tail call i8* @realloc(i8* noundef %19, i64 noundef %20) #50 - because of -   %19 = load i8*          , i8** %18, align 8, !tbaa !2795`

**Comment by ZuseZ4** [comment]
https://fwd.gymni.ch/4vbGQ1

**Comment by ZuseZ4** [comment]
```.ll
; ModuleID = 'ptr12.ll'
source_filename = "ld-temp.o"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

%"class.olb::Cell" = type { [19 x double*] }

declare dso_local i32 @__gxx_personality_v0(...)

; Function Attrs: nofree nosync nounwind readnone speculatable willreturn
declare double @llvm.fmuladd.f64(double, double, double) #0

define hidden { double, double } @foo(%"class.olb::Cell"* noundef nonnull readonly align 8 dereferenceable(168) %arg1) align 2 personality i32 (...)* @__gxx_personality_v0 {
bb:
  %i11 = getelementptr inbounds %"class.olb::Cell", %"class.olb::Cell"* %arg1, i64 0, i32 0, i32 0
  %i12 = load double*, double** %i11, align 8, !tbaa !7
  %i13 = load double, double* %i12, align 8, !tbaa !11
  %i147 = tail call double @llvm.fmuladd.f64(double %i13, double %i13, double %i13) #1
  %i151 = insertvalue { double, double } poison, double 1.000000e+00, 0
  %i152 = insertvalue { double, double } %i151, double %i147, 1
  ret { double, double } %i152
}

declare i8* @_Z17__enzyme_virtualreversePv(...) local_unnamed_addr

define hidden void @main() local_unnamed_addr {
  %1 = tail call i8* (...) @_Z17__enzyme_virtualreversePv({ double, double } (%"class.olb::Cell"*)* nonnull @foo)
  ret void
}

attributes #0 = { nofree nosync nounwind readnone speculatable willreturn }
attributes #1 = { nounwind }

!llvm.ident = !{!0, !0, !0, !0}
!llvm.module.flags = !{!1, !2, !3, !4, !5, !6}

!0 = !{!"clang version 14.0.4 (git@github.com:llvm/llvm-project.git 29f1039a7285a5c3a9c353d054140bf2556d4c4d)"}
!1 = !{i32 1, !"wchar_size", i32 4}
!2 = !{i32 7, !"uwtable", i32 1}
!3 = !{i32 1, !"ThinLTO", i32 0}
!4 = !{i32 1, !"EnableSplitLTOUnit", i32 1}
!5 = !{i32 7, !"frame-pointer", i32 2}
!6 = !{i32 1, !"LTOPostLink", i32 1}
!7 = !{!8, !8, i64 0}
!8 = !{!"any pointer", !9, i64 0}
!9 = !{!"omnipotent char", !10, i64 0}
!10 = !{!"Simple C++ TBAA"}
!11 = !{!12, !12, i64 0}
!12 = !{!"double", !9, i64 0}
```

**Comment by ZuseZ4** [comment]
https://fwd.gymni.ch/WjTVNN
```.ll
; ModuleID = 'ptr12.ll'
source_filename = "ld-temp.o"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

declare dso_local i32 @__gxx_personality_v0(...)

define hidden { double, double } @foo2(double %i13) align 2 personality i32 (...)* @__gxx_personality_v0 {
bb:
  %i151 = insertvalue { double, double } poison, double 1.000000e+00, 0
  %i152 = insertvalue { double, double } %i151, double %i13, 1
  ret { double, double } %i152
}

declare i8* @_Z17__enzyme_virtualreversePv(...) local_unnamed_addr

define hidden void @main() local_unnamed_addr {
  %1 = tail call i8* (...) @_Z17__enzyme_virtualreversePv({ double, double } (double)* nonnull @foo2)
  ret void
}

attributes #0 = { nofree nosync nounwind readnone speculatable willreturn }
attributes #1 = { nounwind }

!llvm.ident = !{!0, !0, !0, !0}
!llvm.module.flags = !{!1, !2, !3, !4, !5, !6}

!0 = !{!"clang version 14.0.4 (git@github.com:llvm/llvm-project.git 29f1039a7285a5c3a9c353d054140bf2556d4c4d)"}
!1 = !{i32 1, !"wchar_size", i32 4}
!2 = !{i32 7, !"uwtable", i32 1}
!3 = !{i32 1, !"ThinLTO", i32 0}
!4 = !{i32 1, !"EnableSplitLTOUnit", i32 1}
!5 = !{i32 7, !"frame-pointer", i32 2}
!6 = !{i32 1, !"LTOPostLink", i32 1}
!7 = !{!8, !8, i64 0}
!8 = !{!"any pointer", !9, i64 0}
!9 = !{!"omnipotent char", !10, i64 0}
!10 = !{!"Simple C++ TBAA"}
!11 = !{!12, !12, i64 0}
!12 = !{!"double", !9, i64 0}
```

**Comment by ZuseZ4** [comment]
closed by #842
